test/integration/data-connect.spec.ts

```diff
@@ -145,13 +164,154 @@ describe('getDataConnect()', () => {
       expect(resp.data.users[0].address).to.be.not.undefined;
     });
 
     it('executeGraphqlRead() should throw for a GraphQL mutation', async () => {
       return getDataConnect(connectorConfig).executeGraphqlRead(mutation)
-        .should.eventually.be.rejected;
+        .should.eventually.be.rejected.and.have.property('code', 'data-connect/permission-denied');
+    });
+  });
+
+  describe('Impersonation', () => {
+    const optsAuthorizedClaims: GraphqlOptions<undefined> = {
+      impersonate: {
+        authClaims: {
+          sub: userId,
+          email_verified: true
+        }
+      }
+    };
+
+    const optsNonExistingClaims: GraphqlOptions<undefined> = {
+      impersonate: {
+        authClaims: {
+          sub: 'non-exisiting-id',
+          email_verified: true
+        }
+      }
+    };
+
+    const optsUnauthorizedClaims: GraphqlOptions<undefined> = {
+      impersonate: {
+        unauthenticated: true
+      }
+    };
+
+    describe('USER Auth Policy', () => {
+      it('executeGraphqlRead() successfully executes an impersonated query with authenticated claims', async () => {
+        const resp =
+          await getDataConnect(connectorConfig).executeGraphqlRead<UsersResponse, undefined>(
+            queryListUsersImpersonation, optsAuthorizedClaims);
+        expect(resp.data.users).to.be.not.empty;
+        expect(resp.data.users.length).equals(1);
+        expect(resp.data.users[0].uid).equals(userId);
+        expect(resp.data.users[0].name).to.not.be.undefined;
+        expect(resp.data.users[0].address).to.not.be.undefined;
+      });
+
+      it('executeGraphqlRead() should throw for impersonated query with unauthenticated claims', async () => {
+        return getDataConnect(connectorConfig).executeGraphqlRead(queryListUsersImpersonation, optsUnauthorizedClaims)
+          .should.eventually.be.rejected.and.have.property('code', 'data-connect/unauthenticated');
+      });
+
+      it('executeGraphql() successfully executes an impersonated query with authenticated claims', async () => {
+        const resp =
+          await getDataConnect(connectorConfig).executeGraphqlRead<UsersResponse, undefined>(
+            queryListUsersImpersonation, optsAuthorizedClaims);
+        expect(resp.data.users).to.be.not.empty;
+        expect(resp.data.users.length).equals(1);
+        expect(resp.data.users[0].uid).equals(userId);
+        expect(resp.data.users[0].name).to.be.not.undefined;
+        expect(resp.data.users[0].address).to.be.not.undefined;
+      });
+
+      it('executeGraphql() should throw for impersonated query with unauthenticated claims', async () => {
+        return getDataConnect(connectorConfig).executeGraphql(queryListUsersImpersonation, optsUnauthorizedClaims)
+          .should.eventually.be.rejected.and.has.property('code', 'data-connect/unauthenticated');
+      });
+
+      it('executeGraphql() successfully executes an impersonated query with non-existing authenticated claims',
```

Comment:
nit: maybe reword this to: "should return an empty list for an impersonated query with non-existing authenticated claims"